cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/kv/HBaseKVTableDefinition.java

```diff
@@ -70,23 +71,23 @@ public DatasetAdmin getAdmin(DatasetSpecification spec, ClassLoader classLoader)
   @Override
   public NoTxKeyValueTable getDataset(DatasetSpecification spec,
                                       Map<String, String> arguments, ClassLoader classLoader) throws IOException {
-    return new KVTableImpl(spec.getName(), hConf);
+    return new KVTableImpl(spec.getName(), hConf, tableUtil);
```

Comment:
Yeah, the way namespaces is implemented in Datasets today, we did not add *namespaceId* to ``DatasetSpecification``. Like we did not add it in ``ApplicationSpecification``. The thought being that when you get a spec, it is already 'in' a namespace. Also, when we store it, the same spec can be stored in two different namespaces with different keys. 
However, I did not consider then that we only have access to the spec in the part of code that translates a dataset instance to an HBase table. Ideally, these methods should get a Wrapper class that contains a namespaceId and ``DatasetSpecification``. However, that change is likely to be pretty big, so extracting the namespaceId from the name right now. As part of the jira I have mentioned, I'll investigate how big of a change it is to do this the right way.